src/LinuxCaptureService/LinuxCaptureServiceBase.cpp

```diff
@@ -149,6 +157,85 @@ class ProducerEventProcessorHijackingFunctionEntryExitForLinuxTracing
   TracingHandler* tracing_handler_;
 };
 
+absl::flat_hash_set<std::string> FileNamesOfAllMinidumps() {
+  absl::flat_hash_set<std::string> result;
+  const std::string kCoreDirectory = "/usr/local/cloudcast/core";
+  auto error_or_core_files = orbit_base::ListFilesInDirectory(kCoreDirectory);
+  if (!error_or_core_files.has_error()) {
+    const std::regex check_if_minidump_file(absl::StrFormat(R"(.*\.core\.dmp)"));
+    for (const std::filesystem::path& path : error_or_core_files.value()) {
+      if (regex_match(path.string(), check_if_minidump_file)) {
+        result.insert(path.string());
+      }
+    }
+  }
+  return result;
+}
+
+struct TargetProcessStateAfterCapture {
+  CaptureFinished::ProcessState process_state;
+  CaptureFinished::TerminationSignal termination_signal;
+};
+
+TargetProcessStateAfterCapture GetTargetProcessStateAfterCapture(
+    pid_t pid, const absl::flat_hash_set<std::string>& old_core_files) {
+  TargetProcessStateAfterCapture result;
+  result.process_state = CaptureFinished::kProcessStateInternalError;
+  result.termination_signal = CaptureFinished::kTerminationSignalInternalError;
+
+  const std::string pid_dir_name = absl::StrFormat("/proc/%i", pid);
+  auto exists_or_error = orbit_base::FileExists(pid_dir_name);
+  if (exists_or_error.has_error()) {
+    ORBIT_ERROR("Unable to check for existence of \"%s\": %s", pid_dir_name,
+                exists_or_error.error().message());
+    // We can't read the process state, so we return an error state.
+    return result;
+  }
+
+  if (exists_or_error.value()) {
+    // Process is still running.
+    result.process_state = CaptureFinished::kRunning;
+    result.termination_signal = CaptureFinished::kTerminationSignalUnspecified;
+    return result;
+  }
+
+  // Check whether we find a minidump. Otherwise we assume the process ended gracefully.
+  const std::string kCoreDirectory = "/usr/local/cloudcast/core";
+  auto error_or_core_files = orbit_base::ListFilesInDirectory(kCoreDirectory);
+  if (error_or_core_files.has_error()) {
+    // We can't access the directory with the core files; we return an error state.
+    return result;
+  }
+  const std::vector<std::filesystem::path>& core_files = error_or_core_files.value();
+  // Matches zero or more characters ('.*'), followed by a literal dot ('\.'), followed by the pid
+  // of the process ('%i'), followed by another literal dot ('\.'), followed by one or more numbers
+  // ('[0-9]+') which is the number of seconds since the epoch finally, followed by the format
+  // ending ('.core.dmp').
+  const std::regex check_if_minidump_file(absl::StrFormat(R"(.*\.%i\.[0-9]+\.core\.dmp)", pid));
+  for (const std::filesystem::path& path : core_files) {
+    if (old_core_files.contains(path.string())) {
```

Comment:
I think we should add a comment (here or elsewhere) explaining why we use `old_core_files`.

Reply:
done